codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyFrameCodec.java

```diff
@@ -95,16 +95,26 @@ public SpdyFrameCodec(
                 SpdyHeaderBlockEncoder.newInstance(version, compressionLevel, windowBits, memLevel), validateHeaders);
     }
 
-    protected SpdyFrameCodec(SpdyVersion version, int maxChunkSize,
-            SpdyHeaderBlockDecoder spdyHeaderBlockDecoder, SpdyHeaderBlockEncoder spdyHeaderBlockEncoder,
-            boolean validateHeaders) {
-        spdyFrameDecoder = new SpdyFrameDecoder(version, this, maxChunkSize);
-        spdyFrameEncoder = new SpdyFrameEncoder(version);
+    protected SpdyFrameCodec(SpdyVersion version,
+                             int maxChunkSize,
+                             SpdyHeaderBlockDecoder spdyHeaderBlockDecoder,
+                             SpdyHeaderBlockEncoder spdyHeaderBlockEncoder,
+                             boolean validateHeaders) {
+        this.spdyFrameDecoder = createDecoder(version, this, maxChunkSize);
+        this.spdyFrameEncoder = createEncoder(version);
         this.spdyHeaderBlockDecoder = spdyHeaderBlockDecoder;
         this.spdyHeaderBlockEncoder = spdyHeaderBlockEncoder;
         this.validateHeaders = validateHeaders;
     }
 
+    protected SpdyFrameDecoder createDecoder(SpdyVersion version, SpdyFrameDecoderDelegate delegate, int maxChunkSize) {
+        return new SpdyFrameDecoder(version, delegate, maxChunkSize);
+    }
+
+    protected SpdyFrameEncoder createEncoder(SpdyVersion version) {
```

Comment:
it's kind of odd to have protected methods that are used in the constructor as people might use variables which are not init yet etc... Wouldn't it be better to add another constructor which will just allow to inject the decoder and encoder directly ?

Reply:
Yes, but when you reference the Codec's subclass which implemented the `SpdyFrameDecoderDelegate`'s subClass, and you can't pass `this` before you call `super(...)`.

Reply:
Yeah just noticed this as well :(